PyInstaller/depend/analysis.py

```diff
@@ -103,6 +107,47 @@ def __init__(self, pyi_homepath, user_hook_dirs=None, *args, **kwargs):
             os.path.join(self._homepath, 'PyInstaller', 'loader', 'rthooks.dat')
         )
 
+        self._initialize_modgraph()
+
+    def _initialize_modgraph(self):
+        """
+        Create the module graph and, for Python 3, analyze dependencies for
+        `base_library.zip` (which remain the same for every executable).
+
+        This function might appear weird but is necessary for speeding up
+        test runtime because it allows caching basic ModuleGraph object that
+        gets created for 'base_library.zip'.
+
+        Parameters
+        ----------
+        excludes : list
+            List of the fully-qualified names of all modules to be "excluded" and
+            hence _not_ frozen into the executable.
+        user_hook_dirs : list
+            List of the absolute paths of all directories containing user-defined
+            hooks for the current application or `None` if no such directories were
+            specified.
+
+        Returns
+        ----------
+        PyiModuleGraph
+            Module graph with core dependencies.
+        """
+        logger.info('Initializing module dependency graph...')
+
+        if not is_py2:
+            logger.info('Analyzing base_library.zip ...')
```

Comment:
I'm quite sure, this is done somewhere else again. Please verify and remove any duplicate code.